mlrun/model_monitoring/controller.py

```diff
@@ -71,14 +71,19 @@ def _get_last_analyzed(self) -> Optional[int]:
         except HttpResponseError as err:
             logger.warn(
                 "Failed to get the last analyzed time for this endpoint and application, "
-                "as this is probably the first time this application is running. "
-                "Using the first request time instead.",
+                "as this is probably the first time this application is running. ",
+                "Using the latest between first request time or last update time minus one day instead.",
                 endpoint=self._endpoint,
                 application=self._application,
                 first_request=self._first_request,
+                last_update=self._stop,
                 error=err,
             )
-            return self._first_request
+
+            return max(
+                self._first_request,
+                self._stop - int(datetime.timedelta(days=1).total_seconds()),
```

Comment:
I'm ok with it for now, but note that this `days=1` constant comes with a price:
This means that monitoring time windows longer than one day will not work.
For example, consider an app that expects one-week windows. This logic limits the difference between the "start" (last_analyzed - L85) to "stop" (last_updated) to one day at most. The app will never run, as the first one-week window will not be completed.

FYI @assaf758.

Reply:
Good point when we will have a policies, we will need to change it according to it.

Reply:
I think we can overcome this issue by updating the the KV last_analyzed with value we are returning.